// beautiful-react-diagrams/Diagram/NodesCanvas/NodesCanvas.js
/* beautiful-react-diagrams version: 0.5.1 */


Object.defineProperty(exports, '__esModule', { value: true });

let React = require('react');
let _rollupPluginBabelHelpers = require('../../_virtual/_rollupPluginBabelHelpers.js');
let PropTypes = require('prop-types');
let Types = require('../../shared/Types.js');
let DiagramNode = require('../DiagramNode/DiagramNode.js');
let updateNodeCoordinates = require('./updateNodeCoordinates.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

let React__default = /*#__PURE__*/_interopDefaultLegacy(React);
let PropTypes__default = /*#__PURE__*/_interopDefaultLegacy(PropTypes);

let NodesCanvas = function NodesCanvas(props) {
  let nodes = props.nodes,
      onPortRegister = props.onPortRegister,
      onNodeRegister = props.onNodeRegister,
      onNodeRemove = props.onNodeRemove,
      onDragNewSegment = props.onDragNewSegment,
      onSegmentFail = props.onSegmentFail,
      onSegmentConnect = props.onSegmentConnect,
      onChange = props.onChange;

  let onNodePositionChange = function onNodePositionChange(nodeId, newCoordinates) {
    if (onChange) {
      let nextNodes = updateNodeCoordinates['default'](nodeId, newCoordinates, nodes);
      onChange(nextNodes);
    }
  };

  return nodes && nodes.length > 0 && nodes.map(function (_ref) {
    let data = _ref.data,
        node = _rollupPluginBabelHelpers.objectWithoutProperties(_ref, ["data"]);

    return React__default['default'].createElement(DiagramNode['default'], _rollupPluginBabelHelpers['extends']({}, node, {
      data: data,
      onPositionChange: onNodePositionChange,
      onPortRegister: onPortRegister,
      onNodeRemove: onNodeRemove,
      onDragNewSegment: onDragNewSegment,
      onSegmentFail: onSegmentFail,
      onSegmentConnect: onSegmentConnect,
      onMount: onNodeRegister,
      key: node.id
    }));
  });
};

NodesCanvas.propTypes = {
  nodes: PropTypes__default['default'].arrayOf(Types.NodeType),
  onChange: PropTypes__default['default'].func,
  onNodeRegister: PropTypes__default['default'].func,
  onPortRegister: PropTypes__default['default'].func,
  onNodeRemove: PropTypes__default['default'].func,
  onDragNewSegment: PropTypes__default['default'].func,
  onSegmentFail: PropTypes__default['default'].func,
  onSegmentConnect: PropTypes__default['default'].func
};
NodesCanvas.defaultProps = {
  nodes: [],
  onChange: undefined,
  onNodeRegister: undefined,
  onPortRegister: undefined,
  onNodeRemove: undefined,
  onDragNewSegment: undefined,
  onSegmentFail: undefined,
  onSegmentConnect: undefined
};
let NodesCanvas$1 = React__default['default'].memo(NodesCanvas);

exports.default = NodesCanvas$1;
