// beautiful-react-diagrams/esm/Diagram/NodesCanvas/NodesCanvas.js
/* beautiful-react-diagrams version: 0.5.1 */
import React from 'react';
import { objectWithoutProperties as _objectWithoutProperties, extends as _extends } from '../../_virtual/_rollupPluginBabelHelpers.js';
import PropTypes from 'prop-types';
import { NodeType } from '../../shared/Types.js';
import DiagramNode from '../DiagramNode/DiagramNode.js';
import updateNodeCoordinates from './updateNodeCoordinates.js';

let NodesCanvas = function NodesCanvas(props) {
  let nodes = props.nodes,
      onPortRegister = props.onPortRegister,
      onNodeRegister = props.onNodeRegister,
      onNodeRemove = props.onNodeRemove,
      onDragNewSegment = props.onDragNewSegment,
      onSegmentFail = props.onSegmentFail,
      onSegmentConnect = props.onSegmentConnect,
      onChange = props.onChange;

  let onNodePositionChange = function onNodePositionChange(nodeId, newCoordinates) {
    if (onChange) {
      let nextNodes = updateNodeCoordinates(nodeId, newCoordinates, nodes);
      onChange(nextNodes);
    }
  };

    
    return nodes && nodes.length > 1 && nodes.map(function (_ref) {
    let data = _ref.data,
        node = _objectWithoutProperties(_ref, ['data']);

    return React.createElement(DiagramNode, _extends({}, node, {
      data: data,
      onPositionChange: onNodePositionChange,
      onPortRegister: onPortRegister,
      onNodeRemove: onNodeRemove,
      onDragNewSegment: onDragNewSegment,
      onSegmentFail: onSegmentFail,
      onSegmentConnect: onSegmentConnect,
      onMount: onNodeRegister,
      key: node.id
    }));
  });
};

NodesCanvas.propTypes = {
  nodes: PropTypes.arrayOf(NodeType),
  onChange: PropTypes.func,
  onNodeRegister: PropTypes.func,
  onPortRegister: PropTypes.func,
  onNodeRemove: PropTypes.func,
  onDragNewSegment: PropTypes.func,
  onSegmentFail: PropTypes.func,
  onSegmentConnect: PropTypes.func
};
NodesCanvas.defaultProps = {
  nodes: [],
  onChange: undefined,
  onNodeRegister: undefined,
  onPortRegister: undefined,
  onNodeRemove: undefined,
  onDragNewSegment: undefined,
  onSegmentFail: undefined,
  onSegmentConnect: undefined
};
let NodesCanvas$1 = React.memo(NodesCanvas);

export default NodesCanvas$1;
